polkadot/node/core/pvf/common/src/worker/security/change_root.rs

```diff
@@ -59,20 +57,12 @@ pub fn enable_for_worker(worker_info: &WorkerInfo) -> Result<()> {
 #[cfg(target_os = "linux")]
 pub fn check_is_fully_enabled(tempdir: &Path) -> Result<()> {
 	let worker_dir_path = tempdir.to_owned();
-	match std::thread::spawn(|| {
-		try_restrict(&WorkerInfo {
-			pid: std::process::id(),
-			kind: WorkerKind::CheckPivotRoot,
-			version: None,
-			worker_dir_path,
-		})
+	try_restrict(&WorkerInfo {
+		pid: std::process::id(),
+		kind: WorkerKind::CheckPivotRoot,
+		version: None,
+		worker_dir_path,
```

Comment:
why was this not caught before?

Reply:
What do you mean @alindima?

Reply:
I'm referring to the supposed error when pivoting root in a multi-threaded process

Reply:
Are you referring to the comments? The comment is already presnt in [master](https://github.com/paritytech/polkadot-sdk/blob/cfa19c37e6/polkadot/node/core/pvf/common/src/worker/security/mod.rs#L40), I just added it to the new `pub fn`s in the refactor. This was definitely caught before, it prevents this from being unit-tested in Rust's default test apparatus which is multi-threaded.

Reply:
I was referring to [this commit](https://github.com/paritytech/polkadot-sdk/pull/2486/commits/ec9bd4b52c54ffd408c4c6d8119d0f9fc3b44e95). I got confused because I was reviewing commit-by-commit